data-weave-plugin/src/main/java/org/mule/tooling/lang/dw/preview/PreviewToolWindowPanel.java

```diff
@@ -150,6 +150,7 @@ private void setFile(@Nullable PsiFile psiFile) {
           mainPanel.remove(previewComponent);
           previewComponent = null;
         }
+        weavePreviewComponent = new WeavePreviewComponent(myProject);
```

Comment:
Shouldn't we only re-created if it is disposed? Shouldn't we check that

Reply:
Good point, I'll add a check.

Reply:
A check was added in the latest commit.